README.md

```diff
@@ -1 +1,60 @@
-# ul-image-api
\ No newline at end of file
+# UL Image API
+
+The [Unified Listing](http://ul.gpii.net) is a resource for people who want to find assistive technology solutions and
+mainstream products with assistive features.  This package is used within the Unified Listing to allow:
+
+1. Contributing images and image metadata associated with [Unified Listing](http://ul.gpii.net/) entries.
+2. Reviewing and approving or rejecting contributions.
+3. Allowing users to see the status of their contributions.
+4. Viewing approved images as part of a product listing.
+
+Anyone (even users who are not logged in) can view the images and galleries associated with the "unified" source.  This
+is a curated data source created by reviewers based on imported data and user contributions.
+
+By [signing up for a free account](/api/user/signup) and [logging in](/api/user/login), users gain the ability to
+contribute updates to the data served by this API.
+
+Users can contribute new images, which are only visible to the submitter and reviewers.  An image is approved or
+rejected by a reviewer.
+
+Users can submit metadata regarding an image (for example, to suggest a better description for a given image).
+These suggestions are only visible to the submitter and reviewers.  Reviewers approve and reject contributed metadata.
+
+To contribute image data to the live Unified Listing, please visit [http://ul.gpii.net].  To try out the software
+locally, please read the "Developer Documentation" below.
+
+# Developer Documentation
+
+## The API
+
+The API provides a series of REST inputs.  For details regarding the endpoints, their accepted input, and their return
+values, see [docs/api-rest-docs.md](the API documentation).
+
+The API is built as a series of [Fluid components](http://fluidproject.org/). For more information, see
+[docs/api-component-docs.md](the API component documentation).  In addition, the API is built using the following key
+libraries:
+
+1. [Sharp](http://sharp.dimens.io/), for image resizing.
```

Comment:
Image resizing was one of the stated goals of the work, as we want to support both "thumbnails" and original images.  My starting assumption was that we needed something that supported common image formats, especially those we already have in our collection of images coming from EASTIN.  I prepared a report to confirm our holdings, we have a mixture of PNG, GIF, and JPEG images, with one webp file.  As webp is not widely supported within browsers, I considered that an optional requirement.

When looking for a library, I did a number of google searches and npm searches, and looked through a range of candidates.  A few names kept coming up, all of which are mentioned in the excellent [performance docs for "sharp"](http://sharp.dimens.io/en/stable/performance/):

1. [sharp](http://sharp.dimens.io/)
2. [lwip](https://www.npmjs.com/package/lwip)
3. [jimp](https://www.npmjs.com/package/jimp)
4. [imagemagick](https://www.npmjs.com/package/imagemagick-native)

Of the pure javascript approaches, sharp was the fastest by far and gave more of an impression of maturity as a project/community than "jimp" or "lwip".  By that I mean that the docs were more complete, the community was a healthy size, the release cadence seemed healthy, and the API seemed well though out.  Imagemagick is a classic that I used productively in the late 90s, but it is slower, less well maintained, and would increase the deployment and test complexity without adding any real value.

In summary, sharp is by far the fastest option I could find, and my impression was that it was the best starting point for image resizing.